src/H5ESlist.c

```diff
@@ -88,6 +88,8 @@ H5ES__list_append(H5ES_event_list_t *el, H5ES_event_t *ev)
     HDassert(el);
     HDassert(ev);
 
+    ev->next = NULL;
```

Comment:
When we append an event to an event set, make sure to set the 'next' pointer to NULL, or it could be pointing to garbage, an event that's been freed, etc. This was problematic when moving events from the event set's active list to its failed list, where list iteration on failed events would previously try to follow into event nodes that were on the active list or were removed from the event set after completion, but a failed event node still pointed to them.

Reply:
we should really stop manipulating lists like that anyway, i.e. and setting head and tails manually. I'll talk to you offline about it.